### PR TITLE
Add dotnet-monitor .NET tool documentation

### PR DESCRIPTION
**Pull request by jander-msft**
## Summary

Adds basic command line documentation for the dotnet-monitor diagnostic tool.
Updates TOC and other related diagnostic pages to link to this new documentation.

This basic documentation does reference to the fact that the tool has an HTTP API as well as some configuration elements, both of which I would like to propose to add documentation in subsequent pull requests. I would like to add sample usage and tutorials in subsequent pull requests as well.

cc @IEvangelist, @alexwolfmsft, @ldillonel, @poppastring 
cc @dotnet/dotnet-monitor, @dotnet/dotnet-diag

### CODE REVIEW

File: docs/core/diagnostics/dotnet-monitor.md [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+
+  It is strongly recommended that this option is not used in production environments.
+  
+- **`--temp-apikey`**
+
+  Generates a temporary API key for the `dotnet-monitor` instance.
+
+- **`--no-http-egress`**
+
+  Disables egress of diagnostic artifacts via the HTTP response. When specified, artifacts must be egress using an egress provider.
```
Comment (kkeirstead) [inline docs/core/diagnostics/dotnet-monitor.md:107]:
```suggestion
  Disables egress of diagnostic artifacts via the HTTP response. When specified, artifacts must be egressed using an egress provider.
```

File: docs/core/diagnostics/dotnet-monitor.md [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+
+- **`-u|--urls <urls>`**
+
+  Bindings for the HTTP api. Default is `https://localhost:52323`.
+
+  This value is mapped into configuration as the `urls` key.
+
+- **`-m|--metrics [true|false]`**
+
+  Enable publishing of metrics to `/metrics` route. Default is `true`
```
Comment (kkeirstead) [inline docs/core/diagnostics/dotnet-monitor.md:133]:
```suggestion
  Enable publishing of metrics to `/metrics` route. Default is `true`.
```

File: docs/core/diagnostics/dotnet-monitor.md [hunk 3]
```diff
@@ -0,0 +1,214 @@
...
+  
+- **`--temp-apikey`**
+
+  Generates a temporary API key for the `dotnet-monitor` instance.
+
+  This value is mapped into configuration as the `Authentication:MonitorApiKey` key.
+
+- **`--no-http-egress`**
+
+  Disables egress of diagnostic artifacts via the HTTP response. When specified, artifacts must be egress using an egress provider.
```
Comment (kkeirstead) [inline docs/core/diagnostics/dotnet-monitor.md:169]:
```suggestion
  Disables egress of diagnostic artifacts via the HTTP response. When specified, artifacts must be egressed using an egress provider.
```

File: docs/core/diagnostics/dotnet-monitor.md [hunk 4]
```diff
@@ -0,0 +1,214 @@
...
+
+  This value is not mapped into configuration.
+
+- **`--level`**
+
+  Configuration level. `Full` configuration can show sensitive information. There are two levels:
+
+  - `Full` - The full configuration without any redaction of any values.
+  - `Redacted` - The full configuration but sensitive information, such as know secrets, is redacted.
```
Comment (kkeirstead) [inline docs/core/diagnostics/dotnet-monitor.md:178]:
```suggestion
  - `Redacted` - The full configuration but sensitive information, such as known secrets, is redacted.
```

File: docs/core/diagnostics/diagnostics-client-library.md [hunk 5]
```diff
@@ -10,7 +10,7 @@ ms.author: tommcdon
...
+Microsoft.Diagnostics.NETCore.Client (also known as the Diagnostics client library) is a managed library that lets you interact with .NET Core runtime (CoreCLR) for various diagnostics related tasks, such as tracing via [EventPipe](eventpipe.md), requesting a dump, or attaching an ICorProfiler. This library is the backing library behind many diagnostics tools such as [`dotnet-counters`](dotnet-counters.md), [`dotnet-trace`](dotnet-trace.md), [`dotnet-gcdump`](dotnet-gcdump.md), [`dotnet-dump`](dotnet-dump.md), and [`dotnet-monitor`](dotnet-monitor.md). Using this library, you can write your own diagnostics tools customized for your particular scenario.
```
Comment (IEvangelist) [inline docs/core/diagnostics/diagnostics-client-library.md:13]:
```suggestion
`Microsoft.Diagnostics.NETCore.Client` (also known as the Diagnostics client library) is a managed library that lets you interact with .NET Core runtime (CoreCLR) for various diagnostics related tasks, such as tracing via [EventPipe](eventpipe.md), requesting a dump, or attaching an `ICorProfiler`. This library is the backing library behind many diagnostics tools such as [dotnet-counters](dotnet-counters.md), [dotnet-trace](dotnet-trace.md), [dotnet-gcdump](dotnet-gcdump.md), [dotnet-dump](dotnet-dump.md), and [dotnet-monitor](dotnet-monitor.md). Using this library, you can write your own diagnostics tools customized for your particular scenario.
```
Inline code is not rendered as such within links, let's remove that.

File: docs/core/diagnostics/dotnet-monitor.md [hunk 6]
```diff
@@ -0,0 +1,214 @@
...
+
+  Shows command-line help.
+
+## Commands
+
+| Command                                                   |
+| --------------------------------------------------------- |
+| [dotnet monitor collect](#dotnet-monitor-collect)         |
+| [dotnet monitor config show](#dotnet-monitor-config-show)    |
+| [dotnet monitor generatekey](#dotnet-monitor-generatekey) |
```
Comment (IEvangelist) [inline docs/core/diagnostics/dotnet-monitor.md:57]:
```suggestion
| [dotnet monitor config show](#dotnet-monitor-config-show) |
| [dotnet monitor generatekey](#dotnet-monitor-generatekey) |
```

**Review by IEvangelist — COMMENTED** [review]
A few nits

**Review by IEvangelist — APPROVED** [review]

**Pull request merged**